patchwork-enum-hacks/src/main/java/com/patchworkmc/mixin/enumhacks/EnchantmentTargetInnerMixin.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.mixin.enumhacks;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.util.function.Predicate;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import net.minecraft.item.Item;
+
+import com.patchworkmc.impl.enumhacks.PatchworkEnchantmentTarget;
+
+@Mixin(targets = "net.minecraft.enchantment.EnchantmentTarget$1")
+public class EnchantmentTargetInnerMixin implements PatchworkEnchantmentTarget {
+	@Unique
+	private boolean isPatchwork = false;
+
+	@Unique
+	private Predicate<Item> predicate;
+
+	// Have to use full signature or the annotation processor complains

```

Comment:
Generally we *should* be using the full signature for all mixin method targets, unfortunately we haven't really done that a lot of the time. Perhaps all the methods in this PR could use it as a start...